### PR TITLE
[MIRROR] makes it so the sanitize_hexcolors' default is 6 characters rather than 3 and gets rid of color_legacy [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/61980
--------------------
## About The Pull Request

title

I thoroughly tested this on tg code, and unless I missed something, which, I'm gonna admit is totally possible, it should be totally fine, even with legacy prefs.

Unless you know of a reason as to why it was 3 characters rather than 6 in which case please tell me

## Why It's Good For The Game

***better colour picking***

## Changelog

:cl:
code: removes color_legacy prefs and changes sanitize_hexcolor()'s default to 6 characters
/:cl:

### CODE REVIEW

**Comment by Seris02** [comment]
see #8692

I'll update it a bit to include any skyrat only ones when I have time

**Comment by Seris02** [comment]
aight, everything is done in the pr referenced above so uh, do what you do, I guess

might wanna make another prefs backup ***just in case***

File: code/modules/client/preferences/species_features/mutants.dm [hunk 1]
```diff
@@ -1,21 +1,25 @@
+<<<<<<< HEAD
 /* SKYRAT EDIT REMOVAL
 /datum/preference/color_legacy/mutant_color
+=======
+/datum/preference/color/mutant_color
+>>>>>>> 694c2999b08 (makes it so the sanitize_hexcolors' default is 6 characters rather than 3 and gets rid of color_legacy (#61980))
```
Comment (GoldenAlpharex) [inline code/modules/client/preferences/species_features/mutants.dm]:
```suggestion
/* SKYRAT EDIT REMOVAL
/datum/preference/color/mutant_color

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 2]
```diff
@@ -962,9 +962,14 @@
...
 		if(draw_color)
-			limb.color = "#[draw_color]"
+			limb.color = draw_color
 			if(aux_zone)
+<<<<<<< HEAD
 				aux.color = "#[draw_color]"
+=======
+				aux.color = draw_color
+
+>>>>>>> 694c2999b08 (makes it so the sanitize_hexcolors' default is 6 characters rather than 3 and gets rid of color_legacy (#61980))
```
Comment (GoldenAlpharex) [inline code/modules/surgery/bodyparts/_bodyparts.dm]:
```suggestion
				aux.color = draw_color
```

File: code/__HELPERS/mobs.dm [hunk 3]
```diff
@@ -80,10 +80,14 @@
...
 */
 //SKYRAT EDIT REMOVAL END
 
+=======
+	return(list("mcolor" = "#[pick("7F","FF")][pick("7F","FF")][pick("7F","FF")]","ethcolor" = GLOB.color_list_ethereal[pick(GLOB.color_list_ethereal)], "tail_lizard" = pick(GLOB.tails_list_lizard), "tail_human" = "None", "wings" = "None", "snout" = pick(GLOB.snouts_list), "horns" = pick(GLOB.horns_list), "ears" = "None", "frills" = pick(GLOB.frills_list), "spines" = pick(GLOB.spines_list), "body_markings" = pick(GLOB.body_markings_list), "legs" = "Normal Legs", "caps" = pick(GLOB.caps_list), "moth_wings" = pick(GLOB.moth_wings_list), "moth_antennae" = pick(GLOB.moth_antennae_list), "moth_markings" = pick(GLOB.moth_markings_list), "tail_monkey" = "None"))
+>>>>>>> 694c2999b08 (makes it so the sanitize_hexcolors' default is 6 characters rather than 3 and gets rid of color_legacy (#61980))
```
Comment (GoldenAlpharex) [inline code/__HELPERS/mobs.dm]:
```suggestion
	return(list("mcolor" = "#[pick("7F","FF")][pick("7F","FF")][pick("7F","FF")]","ethcolor" = GLOB.color_list_ethereal[pick(GLOB.color_list_ethereal)], "tail_lizard" = pick(GLOB.tails_list_lizard), "tail_human" = "None", "wings" = "None", "snout" = pick(GLOB.snouts_list), "horns" = pick(GLOB.horns_list), "ears" = "None", "frills" = pick(GLOB.frills_list), "spines" = pick(GLOB.spines_list), "body_markings" = pick(GLOB.body_markings_list), "legs" = "Normal Legs", "caps" = pick(GLOB.caps_list), "moth_wings" = pick(GLOB.moth_wings_list), "moth_antennae" = pick(GLOB.moth_antennae_list), "moth_markings" = pick(GLOB.moth_markings_list), "tail_monkey" = "None"))
	*/
	//SKYRAT EDIT REMOVAL END
```

File: code/modules/admin/create_mob.dm [hunk 4]
```diff
@@ -16,19 +16,23 @@
...
 	H.dna.blood_type = random_blood_type()
 
 	// Mutant randomizing, doesn't affect the mob appearance unless it's the specific mutant.
+<<<<<<< HEAD
 	//SKYRAT EDIT REMOVAL BEGIN - CUSTOMIZATION
 	/*
 	H.dna.features["mcolor"] = random_short_color() SKYRAT EDIT - We dont use those in favor for a more modular system
+=======
+	H.dna.features["mcolor"] = "#[random_color()]"
+>>>>>>> 694c2999b08 (makes it so the sanitize_hexcolors' default is 6 characters rather than 3 and gets rid of color_legacy (#61980))
```
Comment (GoldenAlpharex) [inline code/modules/admin/create_mob.dm]:
```suggestion
	//SKYRAT EDIT REMOVAL BEGIN - CUSTOMIZATION
	/*
	H.dna.features["mcolor"] = "#[random_color()]"
```

**Comment by GoldenAlpharex** [comment]
Yeet, let's not update until I merged Seris' other PR.

**Pull request merged**